### ISSUE TITLE
No reason crash by adding 4 lines (cpc)

### ISSUE THREAD

**Issue by ldir-hector** [spotlessmind1975/ugbasic#555]
I have a working program for the amstrad CPC in this attached file
[trajectoireCollision.zip](https://github.com/spotlessmind1975/ugbasic/files/10438761/trajectoireCollision.zip)

when I uncomment theses lines (68-71)
```
	IF POINT(10,10) = RED THEN
			CLS
			PRINT "KBOOM !"
	ENDIF

```
The amstrad CPC just reboot without doing anything ! I got this problem with many other lines of code I tried.
Have I reached the memory limit ?
[trajectoireCollision.cpc.dsk.zip](https://github.com/spotlessmind1975/ugbasic/files/
![symbolesCPC_mode_1](https://user-images.githubusercontent.com/118670680/212996656-f502df23-d5fa-4963-b23c-51e1ed8ab250.png)
10438786/trajectoireCollision.cpc.dsk.zip)

Here are all the files. I know that this "font files" il quite big, but I really like it !

**Comment by spotlessmind1975** [comment]
Hello @ldir-hector and **thank you for your bug reportiing!**

Could you attach the file used in the example?
So I can try to reproduce the issue.

**Comment by ldir-hector** [comment]
The DSK : 
[trajectoireCollision.cpc.dsk.zip](https://github.com/spotlessmind1975/ugbasic/files/10451472/trajectoireCollision.cpc.dsk.zip)
The Source : 
[trajectoireCollision.zip](https://github.com/spotlessmind1975/ugbasic/files/10451476/trajectoireCollision.zip)
The picture : 
[symbolesCPC_mode_1.zip](https://github.com/spotlessmind1975/ugbasic/files/10451479/symbolesCPC_mode_1.zip)

**Comment by spotlessmind1975** [comment]
Hello @ldir-hector and **thank you for your bug report!**

I confirm that adding those instructions causes the code overflow. I made two changes to allow the program to run. The first change was to move the stack from address $8000 to address $C000. This will allow programs to be approximately 16 KB longer. The second change is to enable basic peephole optimizations for the Z80 CPU. It was a change I wished I'd introduced sooner, but hadn't had time to test it. So I ask you the kindness to check if the program works. If it doesn't work, you can deactivate the optimizations by clicking on the **BUILD > OPTIONS** menu and deactivating the relative option.

![image](https://user-images.githubusercontent.com/11718606/213879787-2d2d96f5-0f10-44f3-aab2-4f5629d5524d.png)

The issue has been published on COLDFIX v1.12.1 [rev. 20230121.1818].

**Comment by ldir-hector** [comment]
On my configuration I only had "Debugging of load image (-d) checked by default. I tried theses settings and I got a different result, during compilation I got a huge warning window : 
![image](https://user-images.githubusercontent.com/118670680/213909996-d5c1912d-1724-4a34-8dc9-4613df6e955c.png)

That's weird, but at least when I launch it,  it doesn't crash and I got a result : 
![image](https://user-images.githubusercontent.com/118670680/213910305-2e7e342f-1d40-4b05-bd95-019c71999824.png)

I am not familiar with memory bank management but I remember that by default on CPC video RAM in BASIC is at &C000. I don't know where it is in the program you generate with ugBasic though.

If I remove all the options in the compiler's options I also get this result : 
![image](https://user-images.githubusercontent.com/118670680/213910526-3bf07d5a-c255-46f6-b2e6-925e66edb6f4.png)

I think that the debugging of LOAD IMAGE is quite memory consuming.
However I realise that my PNG file brings 14336 bytes in memory just to print some numbers on screen. I'll fix that by only load 0..9 characters and keep with the ugBasic characters for the rest. 
"Equivale a imbracciare un martello per schiacciare una mosca." ;) 
I don't know if it is a real expression in italian though !

**Comment by spotlessmind1975** [comment]
Hello @ldir-hector , **thank you for your feedback!**

At the moment it seems to me that the output you get depends on using the "Enable warnings (-W)" option, and it is definitely a bug in the IDE. In one of the next releases it will be corrected, and it will show the warnings exactly like the errors, ie giving evidence of the line and column where it occurred. However, the error seems really strange to me, as if there was no image converter for mode 0. It's really strange!

Currently, ugBASIC allocates video memory at location &HC000, so it shouldn't make a difference. The fact that removing options has the same result implies that options have no effect. 

If you want to change the font, you can use the `FONT LOAD` command. This is actually a preliminary command released on COLDFIX v1.12.1 [rev. 20230123.2302], but it should replace the preloaded fonts, in order to use it instead of the default one. This will save a lot of space, and instead of using PUT IMAGE you can use the PRINT command.

Here you are an example (FONT load reads an 8x8 IMAGES file):

      BITMAP ENABLE
      FONT LOAD "font.png" TO 48 : 48 = '0', 49 = '1', ...
      PRINT "1234"

ps. the expression is "Equivale a sparare a una mosca con un bazooka". :D

**Comment by ldir-hector** [comment]
I have tested this FONT LOAD and it works very well, thanks !
